### PR TITLE
Refactor and Instrument CassandraClient api

### PR DESCRIPTION
**Pull request by fsamuel-bs**
**Goals (and why)**: Use TableReferences whenever possible and instrument with tritium.

**Concerns (what feedback would you like?)**: What else should I add to the API? Should I add the methods that originated the call?

**Priority (whenever / two weeks / yesterday)**: EOD.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2665)

### CODE REVIEW

**Comment by fsamuel-bs** [comment]
kvs-slow-log and tracing to come.

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java [hunk 1]
```diff
@@ -486,15 +486,6 @@ void runOneTimeStartupChecks() {
...
 
-    //todo dedupe this into a name-demangling class that everyone can access
-    protected static String internalTableName(TableReference tableRef) {
-        String tableName = tableRef.getQualifiedName();
-        if (tableName.startsWith("_")) {
-            return tableName;
-        }
-        return tableName.replaceFirst("\\.", "__");
-    }
-
```
Comment (fsamuel-bs) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java]: Drive-by deletion, as this didn't seem to be used anywhere, and this exact method is available on `AbstractKVS.internalTableName`

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientImpl.java [hunk 2]
```diff
@@ -54,17 +57,22 @@ public CassandraClientImpl(Cassandra.Client client) {
...
 
     @Override
-    public List<KeySlice> get_range_slices(ColumnParent column_parent, SlicePredicate predicate, KeyRange range,
+    public List<KeySlice> get_range_slices(TableReference tableRef,
+            SlicePredicate predicate,
+            KeyRange range,
             ConsistencyLevel consistency_level)
             throws InvalidRequestException, UnavailableException, TimedOutException, TException {
-        return client.get_range_slices(column_parent, predicate, range, consistency_level);
+        ColumnParent colFam = new ColumnParent(AbstractKeyValueService.internalTableName(tableRef));
```
Comment (nziebart) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientImpl.java]: nit: refactor this line into a method
